File: airbyte-integrations/connectors/destination-gcs/src/main/java/io/airbyte/integrations/destination/gcs/credential/GcsCredentialConfig.java
/*
 * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.destination.gcs.credential;

import io.airbyte.integrations.destination.s3.credential.BlobStorageCredentialConfig;
import io.airbyte.integrations.destination.s3.credential.S3CredentialConfig;
import java.util.Optional;

public interface GcsCredentialConfig extends BlobStorageCredentialConfig<GcsCredentialType> {

  Optional<S3CredentialConfig> getS3CredentialConfig();

}

File: airbyte-cdk/java/airbyte-cdk/gcs-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/gcs/credential/GcsCredentialConfig.kt
/*
 * Copyright (c) 2026 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.cdk.integrations.destination.gcs.credential

import io.airbyte.cdk.integrations.destination.s3.credential.BlobStorageCredentialConfig
import io.airbyte.cdk.integrations.destination.s3.credential.S3CredentialConfig
import java.util.*

interface GcsCredentialConfig : BlobStorageCredentialConfig<GcsCredentialType> {
    val s3CredentialConfig: Optional<S3CredentialConfig>
}
